### PR TITLE
fix(autonomous_emergency_braking): check if ego_path is empty

### PR DESCRIPTION
**Pull request by beyzanurkaya**
## Description

fixes: https://github.com/autowarefoundation/autoware.universe/issues/6043

## Tests performed

https://github.com/autowarefoundation/autoware.universe/assets/32412808/10856d18-ee94-404b-b5f2-ecd32be82037

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control/autonomous_emergency_braking/src/node.cpp [hunk 1]
```diff
@@ -357,9 +357,27 @@ bool AEB::checkCollision(MarkerArray & debug_markers)
...
+  const double & dt = imu_prediction_time_interval_;
+  const double & horizon = imu_prediction_time_horizon_;
+  for (double t = 0.0; t < horizon + epsilon; t += dt) {
+    current_x = current_x + current_v * std::cos(current_yaw) * dt;
+    current_y = current_y + current_v * std::sin(current_yaw) * dt;
+    current_yaw = current_yaw + current_w * dt;
+
+    current_p.position = tier4_autoware_utils::createPoint(current_x, current_y, 0.0);
+    current_p.orientation = tier4_autoware_utils::createQuaternionFromYaw(current_yaw);
+  }
```
Comment (ismetatabay) [inline control/autonomous_emergency_braking/src/node.cpp:380]: Thank you for the PR :+1: ! I believe there's no need for an implementation like this. This function is invoked when using the calculated predicted path (output of control) without relying on the IMU. Additionally, AEB can check for collisions with either the IMU or Control. Therefore, I think it's enough to:
```cpp
  if (ego_path.empty()) {
    RCLCPP_ERROR_STREAM(get_logger(), "[AEB] ego_path is empty");
    return false;
  }
  ``` 
  This situation occurs when ego goes off lanelet2 map.

Comment (beyzanurkaya) [reply control/autonomous_emergency_braking/src/node.cpp:380]: What I want to do here is to minimize the effect of ego_path on the calculation of other variables, and even if the path is not generated, EGO will still have a current pose. Of course, this current pose will not be useful for any calculations.

Comment (ismetatabay) [reply control/autonomous_emergency_braking/src/node.cpp:380]: @tkimura4-san Can you check this PR when you available? Thanks in advance

**Comment by beyzanurkaya** [comment]
@tkimura4-san could you review this PR?

File: control/autonomous_emergency_braking/src/node.cpp (hunk 1 above)
Comment (tkimura4) [reply control/autonomous_emergency_braking/src/node.cpp:380]: @ismetatabay 
Sorry for the delayed response.

> I believe there's no need for an implementation like this.
 I agree with this.

**Comment by soblin** [comment]
@beyzanurkaya @ismetatabay I believe at least this issue is already fixed in https://github.com/autowarefoundation/autoware.universe/pull/6346. Can you merge/rebase latest branch and them compare ? Thx

**Comment by beyzanurkaya** [comment]
@soblin After I rebased my branch, this problem was solved. I am closing this PR because it is not needed.

**Pull request closed without merging**